### PR TITLE
Bug fontconfig

### PR DESCRIPTION
**Pull request by kserradell**
This PR is related with #2589

### CODE REVIEW

File: var/spack/repos/builtin/packages/fontconfig/package.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ class Fontconfig(AutotoolsPackage):
 
     depends_on('freetype')
     depends_on('libxml2')
-    depends_on('pkg-config', type='build')
+#    depends_on('pkg-config', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/fontconfig/package.py]: I would remove this commit.

**Comment by adamjstewart** [comment]
Can you squash all of your commits down to a single commit? 

Also, if you use the phrase "Fixes #2589" in your first comment, GitHub will automatically close the issue once this PR is merged. See https://help.github.com/articles/closing-issues-via-commit-messages/ for details.

File: var/spack/repos/builtin/packages/fontconfig/package.py (hunk 1 above)
Comment (kserradell) [inline var/spack/repos/builtin/packages/fontconfig/package.py]: Did it. Sorry.

**Comment by kserradell** [comment]
Sorry, @adamjstewart I'm not used to work with git. Should I close the PR and start again?

**Comment by adamjstewart** [comment]
No worries, git is a strange beast. In this case I think you'll want to do a cherry-pick. Check out our git tips and tricks at http://spack.readthedocs.io/en/latest/contribution_guide.html#git-workflows. @davydden probably knows how to do this better though. I've only learned git well enough not to make mistakes, so now I don't really know how to recover from mistakes 😆

**Comment by davydden** [comment]
>  In this case I think you'll want to do a cherry-pick

I think you first need to go to your local `develop` and do `git pull` to make sure it's up-to-date w.r.t. the one at LLNL (of course, assuming that you set it up to track LLNL's `develop` branch!).
If that's not the case, you need to add a remote and make your `develop` track a proper  branch by:
```
git remote add llnl https://github.com/LLNL/spack.git
git fetch llnl develop
git branch --set-upstream-to llnl/develop
```

Then go back to your branch with this PR (`bug-fontconfig`) and do 
```
git rebase develop
```
See http://spack.readthedocs.io/en/latest/contribution_guide.html#rebasing

But at the end of the day, @tgamblin (or anyone else who can merge), can also squash it **during the merge**. So if you don't feel confident trying the above -- don't.

**Comment by becker33** [comment]
We have recently adopted the policy of squashing all commits in merged branches unless we have some specific reason to believe that the git history will be more legible with the commits separate, or that some important piece of historical information would be lost without the individual commits.

**Comment by kserradell** [comment]
Thank you everybody! I'll try to apply all these tips in the next PR.

**Pull request merged**